design/one-pager-resource-connectivity-mvp.md

```diff
@@ -0,0 +1,653 @@
+
+* Add support for creating a new VPC network via a Crossplane VPC `Network`
+  managed resource.
+* Add support for `CloudSQLInstance` managed resources and their resource
+  classes to specify the external resource's `.ipConfiguration.privateNetwork`,
+  and specify the external name of a `Network` managed resource.
+* Add support for `GKECluster` managed resources and their resource classes to
+  specify the [GKE cluster external resource]'s `.network` field, and specify
+  the external name of a `Network` managed resource.
+* Use the `GKECluster` managed resource's existing support for setting the
+  [GKE cluster external resource]'s `.ipAllocationPolicy.useIpAliases` and
+  `.ipAllocationPolicy.createSubnetwork` fields to ensure an appropriate VPC
+  native subnet is automatically created and used.
+
+To add a generous amount of flexibility Crossplane could:
+
+* Add support for creating a new VPC network via a Crossplane VPC `Network`
+  managed resource.
+* Add support for creating a new VPC subnetwork via a VPC `Subnetwork` managed
+  resource.
+* Add support for `CloudSQLInstance` managed resources and their resource
+  classes to specify the external resource's `.ipConfiguration.privateNetwork`,
+  and specify the external name of a `Network` managed resource.
+* Add support for `GKECluster` managed resources to specify the [GKE cluster
+  external resource]'s `.network` and `.subnetwork` fields.
+* Add support for `GKECluster` managed resources and their resource classes to
+  specify the [GKE cluster external resource]'s `.network` field and
+  `.subnetwork` fields, and specify the external names of a `Network` and a
+  `Subnetwork` managed resource.
+* Use the `GKECluster` managed resource's existing support for setting the
+  [GKE cluster external resource]'s `.ipAllocationPolicy.useIpAliases` field.
+
+High-fidelity Crossplane managed resource representations of the aforementioned
+network and subnetwork external resources would look as follows:
+
+```yaml
+---
+apiVersion: vpc.gcp.crossplane.io/v1alpha1
+kind: Network
+metadata:
+  namespace: crossplane-system
+  name: example
+spec:
+  nameFormat: mycoolname
+  description: A really cool VPC network
+  # IPv4Range puts the network in subnetwork-less 'legacy mode'
+  IPv4Range: 10.0.0.0/8
+  # The documentation says this puts the VPC network in auto-create subnet mode.
+  # In my experiments it seemed to create the VPC in legacy mode instead.
+  autoCreateSubnetworks: true
+  peerings:
+  - name: peerwithsomeothercoolnetwork
+    network: someothercoolnetwork
+    autoCreateRoutes: true
+    exchangeSubnetRoutes: true
+  routingConfig:
+    routingMode: REGIONAL
+```
+
+```yaml
+---
+apiVersion: vpc.gcp.crossplane.io/v1alpha1
+kind: Subnetwork
+metadata:
+  namespace: crossplane-system
+  name: example
+spec:
+  nameFormat: mycoolname
+  description: My cool VPC subnetwork
+  network: projects/coolproject/global/networks/coolestvpc
+  ipCidrRange: 192.168.0.0/24
```

Comment:
The subnetwork parts of the GKE cluster are actually super confusing, because there's usually about four ways to do any one thing. It supports auto creating subnets, manually creating subnets, manually allocating CIDRs, automatically allocating CIDRs, etc. In this specific case the values aren't super obvious because they relate to what other CIDRs are already configured. They also require a little knowledge of Kubernetes, i.e. to ensure you have appropriately sized CIDRs for the number of nodes, or pods,  or services you'll need.

That all said, similarly to my comments above think our responsibility here is to provide a high fidelity and unsurprising representation of the underlying cloud API. This way anyone who understands the GKE API understands how to configure the relevant Crossplane managed resource.

https://cloud.google.com/kubernetes-engine/docs/reference/rest/v1beta1/projects.locations.clusters#Cluster has the full documentation.